### PR TITLE
feat!: authorized Katsu endpoints + slightly auth-aware Beacon

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

**Comment by v-rocheleau** [comment]
The migration guide will need to include a section on how to add a public grant for bento-public requests affected.

**Review by v-rocheleau — APPROVED** [review]
LGTM!

**Review by v-rocheleau — APPROVED** [review]
LGTM!

**Comment by davidlougheed** [comment]
@v-rocheleau re-requesting review, since I added more to docs + the new Beacon code needs a bit of testing.

File: docs/migrating_to_17.md [hunk 1]
```diff
@@ -24,24 +26,49 @@ Key points:
...
+```
+
+To create the client secrets for aggregation/Beacon and Grafana (if the latter is enabled), run the following commands:
 
 ```bash
 ./bentoctl.bash start auth
 ./bentoctl.bash init-auth
 ```
 
+Aggregation/Beacon data access authorization will not work until an authorization service grant is configured;
```
Comment (v-rocheleau) [inline docs/migrating_to_17.md:44]: Remind to add the client secrets to the configuration file

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
small docs stuff, otherwise lgtm

File: docs/installation.md [hunk 2]
```diff
@@ -317,24 +319,54 @@ which in Keycloak should be a UUID.
...
-### c. *Optional step:* Assign portal access to all users in the instance realm
+### c. Create a grant for the aggregation and Beacon services
+
+Run the following commands to set up authorization for the aggregation/Beacon client. 
+**Don't forget to replace `<ISSUER_HERE>` with the issuer URL!**
+
+```bash
+# In the future, view:private_portal will need to be removed from this grant.
+bento_authz create grant \
+  '{"iss": "<PUT ISSUER_HERE ISSUER HERE>", "client": "aggregation"}' \
```
[the remark this answers is not in the document]
Comment (davidlougheed) [reply docs/installation.md]: fixed

File: docs/migrating_to_17.md (hunk 1 above)
Comment (davidlougheed) [reply docs/migrating_to_17.md:46]: added

**Review by v-rocheleau — APPROVED** [review]
LGTM!

**Pull request merged**